test/e2e/reconciler/namespace/controller_test.go

```diff
@@ -107,94 +244,43 @@ func TestNamespaceScheduler(t *testing.T) {
-
-			t.Log("Create a new cluster")
-
-			server2RawConfig, err := f.Servers[physicalCluster2Name].RawConfig()
-			require.NoError(t, err, "failed to get server 2 raw config")
-
-			server2Kubeconfig, err := clientcmd.Write(server2RawConfig)
-			require.NoError(t, err, "failed to get server 2 kubeconfig")
-
-			cluster2, err := clusterClient.Create(ctx, &clusterv1alpha1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					GenerateName: "e2e-nss-2-",
-				},
-				Spec: clusterv1alpha1.ClusterSpec{
-					KubeConfig: string(server2Kubeconfig),
-				},
-			}, metav1.CreateOptions{})
-			require.NoError(t, err, "failed to create cluster2")
-
-			t.Log("Delete the old cluster and expect the namespace to end up scheduled to the new cluster.")
-
-			err = clusterClient.Delete(ctx, cluster1.Name, metav1.DeleteOptions{})
-			require.NoError(t, err, "failed to delete cluster1 %q", cluster1.Name)
-
-			err = expect(namespace, scheduledMatcher(cluster2.Name))
-			require.NoError(t, err, "did not see namespace marked scheduled for cluster2 %q", cluster2.Name)
-
-			t.Log("Delete the remaining cluster and expect the namespace to be marked unscheduled.")
+			s := runningServer{
+				RunningServer: server,
+				clusterClient: clusterClient,
+				client:        client,
+				expect:        expect,
+			}
 
-			err = clusterClient.Delete(ctx, cluster2.Name, metav1.DeleteOptions{})
-			require.NoError(t, err, "failed to delete cluster2 %q", cluster2.Name)
+			t.Logf("Set up clients for test after %s", time.Since(start))
+			t.Log("Starting test...")
 
-			err = expect(namespace, unschedulableMatcher())
-			require.NoError(t, err, "did not see namespace marked unschedulable")
+			testCase.work(ctx, t, s)
 		},
-	)
+		)
+	}
 }
 
 type namespaceExpectation func(*corev1.Namespace) error
 
-func unschedulableMatcher() namespaceExpectation {
+func unscheduledMatcher(reason string) namespaceExpectation {
 	return func(object *corev1.Namespace) error {
-		if nscontroller.IsScheduled(object) {
-			return fmt.Errorf("expected an unschedulable namespace, got status.conditions: %#v", object.Status.Conditions)
+		if condition := conditions.Get(&nscontroller.NamespaceConditionsAdapter{Namespace: object}, nscontroller.NamespaceScheduled); condition != nil {
+			if condition.Status == corev1.ConditionTrue {
+				return fmt.Errorf("expected an unscheduled namespace, got status.conditions: %#v", object.Status.Conditions)
+			}
+			if condition.Reason != reason {
+				return fmt.Errorf("expected an unscheduled namespace with reason %s, got status.conditions: %#v", reason, object.Status.Conditions)
+			}
+			return nil
+		} else {
+			return fmt.Errorf("expected an unscheduled namespace, missing scheduled condition: %#v", object.Status.Conditions)
 		}
-		return nil
 	}
 }
 
 func scheduledMatcher(target string) namespaceExpectation {
 	return func(object *corev1.Namespace) error {
-		if !nscontroller.IsScheduled(object) {
+		if scheduled := nscontroller.IsScheduled(object); !scheduled {
```

Comment:
nit: This can be reverted

Reply:
Thanks for the catch! I had reverted half-way, now fully reverted.